app/models/identity.rb

```diff
@@ -2,6 +2,10 @@ class Identity < ActiveRecord::Base
   include NonNullUuid
 
   belongs_to :user
+  belongs_to :sp,
```

Comment:
it looks like in the schema the service_provider is stored as a string. where is this `sp` coming from?